content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/cpp.md

```diff
@@ -23,153 +23,169 @@ further_reading:
-else
-  # Gets the latest release version number from GitHub.
-  get_latest_release() {
-    curl --silent "https://api.github.com/repos/$1/releases/latest" | jq --raw-output .tag_name
-  }
-  DD_OPENTRACING_CPP_VERSION="$(get_latest_release DataDog/dd-opentracing-cpp)"
-  # Download and install dd-opentracing-cpp library.
-  wget https://github.com/DataDog/dd-opentracing-cpp/archive/${DD_OPENTRACING_CPP_VERSION}.tar.gz -O dd-opentracing-cpp.tar.gz
-  mkdir -p dd-opentracing-cpp/.build
-  tar zxvf dd-opentracing-cpp.tar.gz -C ./dd-opentracing-cpp/ --strip-components=1
-  cd dd-opentracing-cpp/.build
-  # Download and install the correct version of opentracing-cpp, & other deps.
-  ../scripts/install_dependencies.sh
-  # Configure the project, build it, and install it.
-  cmake ..
-  make -j
-  make install
-fi
-```
+FetchContent_MakeAvailable(dd-trace-cpp)
+
+# Add `tracer_example` target
+add_executable(tracer_example tracer_example.cpp)
 
-Include `<datadog/opentracing.h>` and create the tracer:
+# Statically link against `dd-trace-cpp`
+# NOTE: To dynamically link against `dd-trace-cpp` use the `dd_trace_cpp_shared` target
+target_link_libraries(tracer_example dd_trace_cpp-static)
+````
 
 ```cpp
 // tracer_example.cpp
-#include <datadog/opentracing.h>
+#include <datadog/span_config.h>
+#include <datadog/tracer.h>
+#include <datadog/tracer_config.h>
+
 #include <iostream>
 #include <string>
 
-int main(int argc, char* argv[]) {
-  datadog::opentracing::TracerOptions tracer_options{"localhost", 8126, "compiled-in example"};
-  auto tracer = datadog::opentracing::makeTracer(tracer_options);
+namespace dd = datadog::tracing;
 
+int main() {
+  dd::TracerConfig config;
+  config.service = "my-service";
+
+  const auto validated_config = dd::finalize_config(config);
+  if (!validated_config) {
+    std::cerr << validated_config.error() << '\n';
+    return 1;
+  }
+
+  dd::Tracer tracer{*validated_config};
   // Create some spans.
   {
-    auto span_a = tracer->StartSpan("A");
-    span_a->SetTag("tag", 123);
-    auto span_b = tracer->StartSpan("B", {opentracing::ChildOf(&span_a->context())});
-    span_b->SetTag("tag", "value");
+    auto span_a = tracer.create_span();
+    span_a.set_name("A");
+    span_a.set_tag("tag", "123");
+    auto span_b = span_a.create_child();
+    span_b.set_name("B");
+    span_b.set_tag("tag", "value");
   }
 
-  tracer->Close();
   return 0;
 }
 ```
 
-Link against `libdd_opentracing` and `libopentracing`, making sure that they are both in your `LD_LIBRARY_PATH`:
-
 ```bash
-g++ -std=c++14 -o tracer_example tracer_example.cpp -ldd_opentracing -lopentracing
-./tracer_example
+cmake -B build .
+cmake --build build --target tracer_example -j
+
+./build/tracer_example
+DATADOG TRACER CONFIGURATION - {"collector":{"config":{"event_scheduler":{"type":"datadog::tracing::ThreadedEventScheduler" ... }}}
 ```
 
-### Dynamic loading
+{{% /tab %}}
+
+{{% tab "Manual" %}}
 
```

Comment:
```suggestion
To manually download and install the dd-trace-cpp library, run the following bash script:
```
^ I think we need to add more introduction to these code examples so it's clear for users. Something like this (please check).